src/main/kotlin/com/example/Employee/services/EmployeeService.kt

```diff
@@ -10,24 +12,44 @@ import java.util.*
 @Service
 public class EmployeeService (private val employeeRepository: EmployeeRepository){
     fun addDetails(employee: Employee){
-        employeeRepository.save(employee)
+        if(employee.age<18){
+            throw InvalidRequestException("Age should be above 18")
+        }
+        else {
+            employeeRepository.save(employee)
+        }
     }
 
     fun get(): List<Employee> {
         return employeeRepository.findAll().toList()
     }
 
     fun getEmployee(employeeId: Int): Employee {
-        return employeeRepository.findById(employeeId).get()
+        return employeeRepository.findById(employeeId).orElseThrow{EntityNotFoundException("employee not found")}
     }
 
     fun deleteEmployee(employeeId: Int){
-        employeeRepository.deleteById(employeeId)
+        val employee = employeeRepository.findById(employeeId)
+        if(employee.isPresent) {
+            employeeRepository.deleteById(employeeId)
+        } else{
+            throw EntityNotFoundException("employee not found")
+        }
     }
 
     fun updateEmployee(employeeId: Int, employee: Employee) {
-        employee.employeeId=employeeId
-        employeeRepository.save(employee)
+        val employeeExists = employeeRepository.findById(employeeId)
```

Comment:
Restructure your code to follow the early return pattern, for easier understanding of code:
https://medium.com/swlh/return-early-pattern-3d18a41bba8